patches/uboot/0002-arm-rpi-copy-rpi-fw-modified-fdt-settings-to-u-boot-.patch

```diff
@@ -0,0 +1,331 @@
++					fdt32_to_cpu(fw_fdt_prop->nameoff));
++			fw_len = fdt32_to_cpu(fw_fdt_prop->len);
++			fw_nodep = fw_fdt_prop->data;
++			if (fw_len < 0) {
++				printf ("libfdt fdt_getprop(): %s\n",
++					fdt_strerror(fw_len));
++				return 1;
++			}
++			/* Try to find a property that has the name full_fw_path. */
++			fdt_prop = fdt_get_property(fdt, nodeoffset,
++					            fw_pathp, &len);
++			pathp = fdt_string(fdt,
++					   fdt32_to_cpu(fdt_prop->nameoff));
++			len = fdt32_to_cpu(fdt_prop->len);
++			nodep = fdt_prop->data;
++
++			/*
++			 * Verify if the property exists in the node, if not
++			 * creates it. Otherwise, verify also if the fw_fdt node
++			 * property data has the same value as the node property
++			 * that has been checked.
++			 */
++			if (len < 0 && len != -FDT_ERR_NOTFOUND) {
++				printf ("libfdt fdt_get_property(): %s\n",
++					fdt_strerror(len));
++				return 1;
++			} else if (len == -FDT_ERR_NOTFOUND ||
++					strcmp(fw_pathp, pathp) != 0 ||
++					fw_len != len ||
++					memcmp(fw_nodep, nodep, len) != 0) {
++				ret = fdt_setprop(fdt, nodeoffset,
++						fw_pathp, fw_nodep, fw_len);
++				if (ret < 0)
++				{
++					printf("libfdt fdt_setprop(): %s\n",
++							fdt_strerror(ret));
++					return 1;
++				}
++			}
++			else if (len < 0) {
++				printf ("libfdt fdt_getprop(): %s\n",
++					fdt_strerror(len));
++				return 1;
++			}
++
++			break;
++		case FDT_NOP:
++			break;
++		case FDT_END:
++			return 1;
++		default:
++			printf("Unknown tag 0x%08X\n", fw_tag);
++			return 1;
++		}
++		fw_nodeoffset = fw_nextoffset;
++	}
++	return 0;
++}
++
++
++static int ft_board_setup_copy_fw_fdt_settings(struct fdt_header * fdt)
++{
++	int ret;
++	int node_num;
++	int nodeoffset;		/* node offset from libfdt */
++	int fw_nodeoffset;
++	int fw_parentoffset;
++	char full_fw_path[50];
++	const char * fw_path;
++	int len;
++	struct fdt_header * fw_fdt = (struct fdt_header *)fw_dtb_pointer;
++
++	if(fdt_totalsize(fw_fdt) > fdt_totalsize(fdt))
++	{
++		ret = fdt_shrink_to_minimum(fdt, fdt_totalsize(fw_fdt) -
++				fdt_totalsize(fdt));
++		if (ret < 0)
++		{
++			printf("Could not resize fdt!\n");
++			return 1;
++		}
++	}
++
++	for (node_num = 0;
++		node_num < ARRAY_SIZE(rpi_firmware_changed_nodes);
++		node_num++)
++	{
++		nodeoffset = fdt_path_offset(fdt,
++				rpi_firmware_changed_nodes[node_num]);
++		fw_nodeoffset = fdt_path_offset(fw_fdt,
++				rpi_firmware_changed_nodes[node_num]);
++		if (fw_nodeoffset == -FDT_ERR_NOTFOUND) {
++			/*
++			 * If node was not found in fw fdt just jump to
++			 * the next one.
++			 */
++			continue;
++		} else if (fw_nodeoffset < 0) {
++			/* Something bad happened. */
++			printf("libfdt fdt_path_offset() returned %s\n",
++				fdt_strerror(nodeoffset));
++			return 1;
++		}
++
++		while (nodeoffset == -FDT_ERR_NOTFOUND) {
++			/* The fw fdt node was not found in u-boot fdt and it
++			 * must be created.
++			 * First found the fw node parent offset.
```

Comment:
s/found/find/

Reply:
Solved! I have made the same mistake at other comments.